src/main/java/org/jenkinsci/plugins/docker/commons/KeyMaterial.java

```diff
@@ -1,55 +1,73 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
 package org.jenkinsci.plugins.docker.commons;
 
 import hudson.EnvVars;
-import hudson.model.AbstractBuild;
-import org.jenkinsci.plugins.docker.commons.impl.CompositeKeyMaterial;
-import org.jenkinsci.plugins.docker.commons.impl.NullKeyMaterial;
 
-import javax.annotation.Nullable;
 import java.io.Closeable;
 import java.io.IOException;
 import java.io.Serializable;
 
 /**
- * Represents a locally extracted credentials information.
- *
- * <p>
- * Whenever you want to fork off docker directly or indirectly, use this object to set up environment variables
- * so that docker will talk to the right daemon.
- *
- * <p>
- * When you are done using the credentials, call {@link #close()} to allow sensitive information to be removed
- * from the disk.
```

Comment:
Right, kinda. I should have left those same pointers on `KeyMaterial` though. Will re-add them.